cortex-m-qemu/src/lang_items.rs

```diff
@@ -1,16 +1,29 @@
 use core::intrinsics;
 
 #[lang = "start"]
-extern "C" fn start(main: fn(),
-                    _argc: isize,
-                    _argv: *const *const u8)
-                    -> isize {
+extern "C" fn start<T>(main: fn() -> T, _argc: isize, _argv: *const *const u8) -> isize
+where
+    T: Termination,
+{
     main();
 
     0
 }
 
+#[lang = "termination"]
+pub trait Termination {
+    fn report(self) -> i32;
+}
+
+impl Termination for () {
+    fn report(self) -> i32 {
+        0
+    }
+}
+
+
 #[lang = "panic_fmt"]
+#[no_mangle]
```

Comment:
Why has this been added? AFAIK it has no effect because rustc will name this symbol rust_begin_unwind.

Reply:
Yes it is superfluous, I'll revert it.